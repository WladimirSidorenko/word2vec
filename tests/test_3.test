#!/bin/sh

##################################################################
# Variables
BIN=${1}
INPUT_0='test_3.0.in'
OUTPUT_0='test_3.0.out'
EXPECTED_0='test_3.0.expected'
TEST_NAME='task_specific_least_sq'

##################################################################
# Test 0
echo '1..1'
${BIN} -ts-least-sq 1 -min-count 0 -train "${INPUT_0}" \
    -output "${OUTPUT_0}" -threads 1
diff -q "${OUTPUT_0}" "${EXPECTED_0}" > /dev/null
if test $? -eq 0; then
    echo 'ok 1 # least square vectors trained'
else
    echo 'not ok 1 # least square vectors differ from expectations'
fi
