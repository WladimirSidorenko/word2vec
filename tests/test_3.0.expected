6 3
</s> 0.031362 0.007044 -0.120117 
hello 0.061148 -0.139631 0.055236 
world -0.009431 -0.014673 -0.126024 
you -0.075502 0.091198 0.114758 
are 0.086137 0.058681 0.065580 
beautiful -0.111670 -0.119888 0.150280 
